com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs

```diff
@@ -27,9 +27,12 @@ static OverridableFrameSettingsArea()
             attributes = new Dictionary<FrameSettingsField, FrameSettingsFieldAttribute>();
             attributesGroup = new Dictionary<int, IOrderedEnumerable<KeyValuePair<FrameSettingsField, FrameSettingsFieldAttribute>>>();
             Type type = typeof(FrameSettingsField);
-            foreach (FrameSettingsField value in Enum.GetValues(type))
+            foreach (string enumName in Enum.GetNames(type))
             {
-                attributes[value] = type.GetField(Enum.GetName(type, value)).GetCustomAttribute<FrameSettingsFieldAttribute>();
+                if (type.GetField(enumName).GetCustomAttribute<ObsoleteAttribute>() != null)
+                    continue;
+
+                attributes[(FrameSettingsField)Enum.Parse(type, enumName)] = type.GetField(enumName).GetCustomAttribute<FrameSettingsFieldAttribute>();
```

Comment:
Since we already have static field FrameSettingsFieldAttribute.s_FrameSettingsAutoNames can we reuse this Dictionary? So we just iterate over s_FrameSettingsAutoNames here which gives us both enumName and Enum.Parse(type, enumName) from its iterator.

Reply:
good point. I would have to make this a static cacheable thing since the ordering of static constructors is not guaranteed !

Reply:
Done. Latest commit should have exactly this. I looked into the post you mentioned, it might be too overkill. Well let me know your thoughts.

Reply:
yea thinking about it editor build always runs on mono and that attribute works for il2cpp so it will not do much sorry